dags/postgres_to_clickhouse_etl.py

```diff
@@ -201,11 +201,90 @@ def get_team_table_sql() -> str:
     """
 
 
-def create_clickhouse_tables() -> None:
-    """Create the organization and team tables in ClickHouse on all nodes."""
-    # Create tables on all nodes
-    run_sql_with_exceptions(get_organization_table_sql(), node_role=NodeRole.ALL)
-    run_sql_with_exceptions(get_team_table_sql(), node_role=NodeRole.ALL)
+def create_database_if_not_exists(context: Optional[Union[OpExecutionContext, AssetExecutionContext]] = None) -> None:
+    """Create the models database in ClickHouse if it doesn't exist."""
+    if context:
+        context.log.info("Creating database 'models' if it doesn't exist...")
+    create_db_sql = "CREATE DATABASE IF NOT EXISTS models"
+
+    try:
+        # Use sync_execute instead of run_sql_with_exceptions to ensure same connection
+        sync_execute(create_db_sql)
+        if context:
+            context.log.info("Database 'models' created/verified successfully")
+    except Exception as e:
+        if "already exists" in str(e).lower():
+            if context:
+                context.log.info("Database 'models' already exists")
+        else:
+            if context:
+                context.log.exception(f"Error creating database: {e}")
+            raise
+
+
+def create_clickhouse_tables(
+    context: Optional[Union[OpExecutionContext, AssetExecutionContext]] = None, force_recreate: bool = False
+) -> None:
+    """Create the organization and team tables in ClickHouse on all nodes.
+
+    Args:
+        context: Execution context for logging
+        force_recreate: If True, drop and recreate tables even if they exist
+    """
+    # First ensure the database exists
+    create_database_if_not_exists(context)
+
+    # Verify database exists by listing databases
+    if context:
+        context.log.info("Verifying database exists...")
+    try:
+        result = sync_execute("SHOW DATABASES")
+        databases = [row[0] for row in result]
+        if context:
+            context.log.info(f"Available databases: {databases}")
+        if "models" not in databases:
+            raise Exception("Database 'models' was not created successfully")
+    except Exception as e:
+        if context:
+            context.log.exception(f"Error verifying database: {e}")
+        raise
```

Comment:
Won't this throw anyway when creating the table

And the database should have already been created anyway based on the block right above this

Reply:
yeah, totally not needed.